include/grpc++/auth_context.h

```diff
@@ -34,15 +34,42 @@
 #ifndef GRPCXX_AUTH_CONTEXT_H
 #define GRPCXX_AUTH_CONTEXT_H
 
+#include <iterator>
 #include <vector>
 
 #include <grpc++/config.h>
 
+struct grpc_auth_context;
```

Comment:
Sure. The only lose is it is not under AuthContext any more. 

I guess I will add an AuthContextPropertyIterator in a public header (or under impl/?) and include it here. The main gain would be having its own .cc file to contain its implementation.